### ISSUE TITLE
`anemic-getter` lint ignores redundant `ξ`/`ρ` chains in EO code

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#1295]
Consider this code:

```
[] > foo
  ^ > f
```

Here, `f` is an anemic getter of `^`: it adds nothing, since `^` is reachable
by its own symbol from the very place where `f` sits. However, the
`anemic-getter` lint stays silent about it.

The reason is in `anemic-getter.xsl`: the lint only looks for
`ξ.<name>`-based references whose `<name>` is a sibling attribute of the same
formation:

```xml
<xsl:for-each select="//o[@name and @name!='φ' and @base and matches(@base, '^ξ\.[^.]+$') and not(o)]">
  <xsl:variable name="ref" select="substring-after(@base, 'ξ.')"/>
  <xsl:if test="../o[@name=$ref]">
```

The parser turns `^ > f` into `<o name="f" base="ξ.ρ"/>`. That base does match
`^ξ\.[^.]+$`, so `$ref` becomes `ρ` — and since no sibling is named `ρ`, the
`xsl:if` never fires and no defect is reported.

The same holds for the rest of that family, none of which is caught today:

| EO | XMIR base | caught now? |
| --- | --- | --- |
| `x > y` (`x` is a sibling) | `ξ.x` | yes |
| `^ > f` | `ξ.ρ` | no |
| `^.^ > f` | `ξ.ρ.ρ` | no |
| `$ > f` | `ξ` | no |

All of them are the very same redundancy the lint exists to catch: a named
object whose whole body is nothing but a name for something that is already
nameable, unchanged, right where the getter is written.

Note that `^.bar > f` (base `ξ.ρ.bar`) is a different story — it reaches out of
the formation for an attribute of the parent, and `redundant-attachment`
already reasons about such references — so it should stay out of scope here.

Let's teach `anemic-getter` to complain about bare `ξ`/`ρ` chains too.

**Comment by yegor256** [comment]
@rultor release, tag is `0.4.7`

**Comment by yegor256** [comment]
@volodya-lombrozo can you release new version please?

**Comment by volodya-lombrozo** [comment]
@yegor256 will be published soon:  https://github.com/objectionary/lints/issues/1312#issuecomment-5421842840

**Comment by 0crat** [comment]
@yegor256 Thanks for closing this issue! You've earned +8 points. Your running score is [+1931](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
